diff --git a/vl/pgm.c b/vl/pgm.c
index 72110a3..4425db7 100644
--- a/vl/pgm.c
+++ b/vl/pgm.c
@@ -233,7 +233,7 @@ vl_pgm_extract_head (FILE* f, VlPgmImage *im)
     return vl_set_last_error(VL_ERR_PGM_INV_META, "Invalid PGM meta information");
   }
 
-  if(! (max_value >= 65536)) {
+  if( max_value >= 65536 ) {
     return vl_set_last_error(VL_ERR_PGM_INV_META, "Invalid PGM meta information");
   }
 
